SAMtags.tex

```diff
@@ -453,6 +453,161 @@ \subsubsection{Color space}
+  {\tt Mm} & i & Base modifications / methylation  \\
+% {\tt MP} & i & Base modification qualities \\
+  {\tt Ml} & i & Base modification probabilities \\
+\end{longtable}
+\end{center}
+
+\subsection{Base modifications}
+
+Base modifications, including base methylation, are represented as a series of edits from the primary unmodified sequence as originally reported by the sequencing instrument.
+This potentially differs to the sequence stored in the main SAM {\sf SEQ} field if the latter has been reverse complemented, in which case SAM {\sf FLAG} 0x10 must be set.
+This means modification positions are also recorded against the original orientation (i.e. starting at the 5' end), and count the original base types.
+
+Each modified base listed also has a quality value associated with it.
+Given the unmodified base already has a phred likelihood, this base modification quality should be interpreted as the likelihood of this modification being correct given an assumption the original call is correct.
+
+\begin{description}
+\item[Mm:Z:\tagregex{([ACGTUN][-+][a-z]+(,[0-9]+)*;)*}]
+\hfill\\
+The first character is the unmodified ``fundamental'' base as reported
+by the sequencing instrument.
+It must be one of {\tt A}, {\tt C}, {\tt G}, {\tt T}, {\tt U} (if RNA) or {\tt N}.
+Note {\tt N} may be used to match any base rather than specifically an {\tt N} call by the sequencing instrument.
+This may be used in situations where the base modification is not a derivation of a standard base type.
+This is followed by either plus or minus indicating the strand the modification was observed on (relative to the original sequenced strand of {\sf SEQ} with plus meaning same orientation\footnote{Hence a tool that may reverse complement sequences does not need to understand how to manipulate the {\tt Mm} and {\tt Ml} tags.}), and one or more base modification symbols.
+This is then followed by a comma separated list of how many unmodified seq bases of the stated base type to skip, stored as a delta to the last and starting with 0 as the first (or next) base, starting from the uncomplemented 5' end of the {\sf SEQ} field.
+This number series is comparable to the numbers in an {\tt MD} tag,
+albeit counting specific base types only and potentially reverse-complemented.
+
+For example {\tt C+m,5,12,0;} tells us there are three 5-Methylcytosine bases in the original orientation {\sf SEQ}.
+The first 5 {\tt C} bases are unmodified and the 6th is modified, as are the 19th (with 12 between the 6th and 19th) and 20th.
+Similarly {\tt G-m,14;} indicates the 15th {\tt G} is a 5-Methylcytosine on the opposite strand (still counting from the 5' end).
+
+This permits modifications to be listed on either strand with the rare potential for both strands to have a modification at the same site.
+
+Note it is permitted for the coordinate list to be empty (for example {\tt Mm:Z:C+m;}), which may be used as an explicit indicator that this base modification is not present.
```

Comment:
Getting down to implementing this output in the ONT Megalodon software. In the modified basecaller the standard output will include a modified base call at every relevant base of the read. I believe this would result in the `Mm` tag: `Mm:Z:C+m,0,0,...,0` (with the number of zeros corresponding to the number of `C`s in a read). Is this correct? If so, would it make sense to allow a special exception for the "all bases include a modified base call" to be listed instead without any positions values (i.e. `Mm:Z:C+m`) instead of this being the special case where no bases are identified as modified?

Reply:
After a bit of further development I have changed my mind and think that the current interpretation of the "empty" MM tag is more consistent with the tag. It also seems that thresholding the probabilities to some extent will be useful and so outputting absolutely all potential modified base probabilities will likely not be a primary use case.